service/controller/v22/resource/bridgezone/resource.go

```diff
@@ -347,53 +345,97 @@ func (r *Resource) EnsureDeleted(ctx context.Context, obj interface{}) error {
 			},
 			HostedZoneId: &intermediateZoneID,
 		}
-		time.Sleep(1 * time.Second)
 		_, err := defaultGuest.ChangeResourceRecordSetsWithContext(ctx, in)
 		if err != nil {
 			return microerror.Mask(err)
 		}
-		time.Sleep(1 * time.Second)
 
 		r.logger.LogCtx(ctx, "level", "debug", "message", "ensured deletion of final zone delegation from intermediate zone")
 	}
 
 	return nil
 }
 
+// findHostedZoneID fetches Route53 hosted zone IDs based on a given name. The
+// implementation fetches up to 100 matching results to find the right one. The
+// bridgezone resource here is only concerned with the hosted zone ID of the
+// hosted zone name provided. The desired ID will always be carried in the first
+// Route53 response as the one we want to fetch is the most accurate and always
+// listed as the first item in the response. This is because of the
+// lexicographical order of the response items as the API documentation puts it.
+// See also
+// https://godoc.org/github.com/aws/aws-sdk-go/service/route53#Route53.ListHostedZonesByName.
+//
+//     Retrieves a list of your hosted zones in lexicographic order.
+//
+// Here is an example to make it clearer. Let's consider the following hosted
+// zone name.
+//
+//     9cvgo.k8s.ginger.eu-central-1.aws.gigantic.io
+//
+// Given this name, findHostedZoneID will receive a response from Route53
+// similar to the following example, containing a single hosted zone carrying
+// its ID.
+//
+//     {
+//       ...
+//       HostedZones: [{
+//         ...
+//         Id: "/hostedzone/Z1A4QS1NDU6NW6",
+//         Name: "9cvgo.k8s.ginger.eu-central-1.aws.gigantic.io.",
+//         ...
+//       }],
+//       ...
+//     }
+//
+// The example above was about a very specific domain name, which list result
+// could only find a single item in the response. Let's consider a less specific
+// domain name as input for findHostedZoneID.
+//
+//     k8s.ginger.eu-central-1.aws.gigantic.io
+//
+// The result from Route53 will again list all the childs within the given
+// domain name. In the example response below there where only two tenant
+// clusters.
+//
+//     {
+//       ...
+//       HostedZones: [{
+//         ...
+//         Id: "/hostedzone/Z1HJGG5VLG8GZH",
+//         Name: "k8s.ginger.eu-central-1.aws.gigantic.io.",
+//         ...
+//       },{
+//         ...
+//         Id: "/hostedzone/Z1KSFLSM1JEQYM",
+//         Name: "0tz6i.k8s.ginger.eu-central-1.aws.gigantic.io.",
+//         ...
+//       },{
+//         ...
+//         Id: "/hostedzone/Z1A4QS1NDU6NW6",
+//         Name: "9cvgo.k8s.ginger.eu-central-1.aws.gigantic.io.",
+//         ...
+//       }],
+//       ...
+//     }
+//
```

Comment:
I would drop this godoc. I spend a lot of time reading it while the change below is rather straight forward. It would be enough for me to have a comment in the code like:

```go
	// Start listing with DNSName set to the desired zone name
 	// to make sure the domain is listed in the first page of results.
	in := &route53.ListHostedZonesByNameInput{
		DNSName: aws.String(name),
	}
```

Reply:
I find it pretty useful and would like to keep it. I also do not read the other huge comment above but it might be interesting at some point. Also, somebody put a lot of effort into it and I can imagine some day it pays back.

Reply:
The huge comment above was written by me. It explains our DNS layout and transitions as opposed  to the one here explaining function working internals. IMO the godoc for a func should ideally describe what is taken and what is returned and not how it is processed.